bson/mgoregistry.go

```diff
@@ -0,0 +1,184 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package bson
+
+import (
+	"errors"
+	"reflect"
+)
+
+var (
+	// ErrSetZero may be returned from a SetBSON method to have the value set to its respective zero value.
+	ErrSetZero = errors.New("set to zero")
```

Comment:
Optional: Consider modifying this name to indicate it's only used with the mgo-compatibility registries. E.g. `ErrMgoSetZero`